projects/knora-ui/src/lib/viewer/values/time-value/time-input/time-input.component.ts

```diff
@@ -0,0 +1,236 @@
+
+  constructor(fb: FormBuilder,
+              @Optional() @Self() public ngControl: NgControl,
+              private fm: FocusMonitor,
+              private elRef: ElementRef<HTMLElement>,
+              @Optional() _parentForm: NgForm,
+              @Optional() _parentFormGroup: FormGroupDirective,
+              _defaultErrorStateMatcher: ErrorStateMatcher) { 
+
+    super(_defaultErrorStateMatcher, _parentForm, _parentFormGroup, ngControl);
+
+    this.dateFormControl = new FormControl({value: null, Validators: [Validators.required]});
+    
+    this.timeFormControl = new FormControl({value: null, Validators: [Validators.required, Validators.pattern(CustomRegex.TIME_REGEX)]});
+
+    this.form = fb.group({
+      date: this.dateFormControl,
+      time: this.timeFormControl
+    });
+
+    fm.monitor(elRef.nativeElement, true).subscribe(origin => {
+      this.focused = !!origin;
+      this.stateChanges.next();
+    });
+
+    if (this.ngControl != null) {
+      this.ngControl.valueAccessor = this;
+    }
+  }
+
+  ngDoCheck() {
+    if (this.ngControl) {
+      this.updateErrorState();
+    }
+  }
+
+  ngOnDestroy() {
+    this.stateChanges.complete();
+  }
+
+  onContainerClick(event: MouseEvent) {
+    if ((event.target as Element).tagName.toLowerCase() != 'input') {
+      this.elRef.nativeElement.querySelector('input').focus();
+    }
+  }
+
+  writeValue(datetime: string | null): void {
+    this.value = datetime;
+  }
+
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+  }
+
+  _handleInput(): void {
+    this.onChange(this.value);
+  }
+
+  // return converted Date obj as a string without the milliseconds
+  userInputToTimestamp(userInput: DateTime): string {
+    let splitTime = userInput.time.split(":");
+    const updateDate = new Date(userInput.date.toCalendarPeriod().periodStart.year,
+                                (userInput.date.toCalendarPeriod().periodStart.month - 1),
+                                userInput.date.toCalendarPeriod().periodStart.day,
+                                Number(splitTime[0]),
+                                Number(splitTime[1])
+    );
+
+    return updateDate.toISOString().split('.')[0]+"Z";
+  }
+
+  // converts and returns a unix timestamp string as an array consisting of a GregorianCalendarDate and a string
+  convertTimestampToDateTime(timestamp: string): DateTime {
+    const calendarDate = new CalendarDate(Number(this.datePipe.transform(timestamp, "y")),
+                                          Number(this.datePipe.transform(timestamp, "M")),
+                                          Number(this.datePipe.transform(timestamp, "d")));
+
+    const date = new GregorianCalendarDate(new CalendarPeriod(calendarDate, calendarDate));
+
+    let time = this.datePipe.transform(timestamp, "HH:mm");
```

Comment:
use `const` instead of `let`. Please use single quotes for strings.

Or just use WebStorm .. ;-)

Reply:
fixed in e78d4cc I really tried to only use single quotes, but some slipped through 😜